autonomy/configurations/base.py

```diff
@@ -56,6 +56,73 @@
 }
 
 
+def is_strict_list(data: List) -> bool:
+    """
+    Check if a data list is an strict list
+
+    The data list contains a mapping object we need to process it as an
+    object containing configurable parameters. For example
+
+    cert_requests:
+      - public_key: example_public_key
+
+    This will get exported as `CONNECTION_NAME_CERT_REQUESTS_0_PUBLIC_KEY=example_public_key`
+
+    Where as
+
+    parameters:
+     - hello
+     - world
+
+     will get exported as `SKILL_NAME_PARAMETERS=["hello", "world"]`
```

Comment:
This still have one limitation, the user will have to be explicit when picking up these variables in the `aea-config.yaml` or any other component config where the overrides are defined. To explain this take this config for example

```yaml
cert_requests:
  - public_key: example_public_key_0
  - public_key: example_public_key_1
```

These will be exported as 

```
CONNECTION_NAME_CERT_REQUESTS_0_PUBLIC_KEY=example_public_key_0
CONNECTION_NAME_CERT_REQUESTS_1_PUBLIC_KEY=example_public_key_1
```

When defining the `aea-config.yaml` we won't know how many sub parameters had been exported for a particular list param so we will have to define `aea-config.yaml` as 

```yaml
cert_requests:
  - public_key: ${str:example_public_key_0}
  - public_key: ${str:example_public_key_1}
```